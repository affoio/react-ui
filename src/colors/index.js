const colors = {
  indigoBlue: {
    800: '#4718b4',
    700: '#623abf',
    600: '#8769ce',
    500: '#ac97dd',
    400: '#d1c5ec',
    200: '#ece7f7',
    300: '#e4ddf4',
    100: '#f6f3fb',
  },
  marigold: {
    800: '#ffca00',
    700: '#ffd226',
    600: '#ffdc59',
    500: '#ffe78c',
    400: '#fff2bf',
    300: '#fff7d9',
    200: '#fff9e5',
    100: '#fffcf2',
  },
  cerise: {
    800: '#e2086e',
    700: '#e62d84',
    600: '#ec5ea1',
    500: '#f290be',
    400: '#f8c1db',
    300: '#fbdaea',
    200: '#fce6f0',
    100: '#fdf2f8',
  },
  black: {
    800: '#000000',
    700: '#262626',
    600: '#595959',
    500: '#8c8c8c',
    400: '#bfbfbf',
    300: '#d9d9d9',
    200: '#e5e5e5',
    100: '#f2f2f2',
  },
  seaweed: {
    800: '#13c64f',
    700: '#36ce69',
    600: '#65da8c',
    500: '#95e5b0',
    400: '#c4f1d3',
    300: '#dcf7e5',
    200: '#e7f9ed',
    100: '#f3fcf6',
  },
  white: '#ffffff',
};

export default colors;
