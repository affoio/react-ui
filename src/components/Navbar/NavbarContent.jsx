import styled from 'styled-components';

export const NavbarContent = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
`;
