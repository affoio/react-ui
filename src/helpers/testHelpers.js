import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from 'enzyme';

import { MemoryRouter } from 'react-router';

export const renderWithThemeProvider = (children, theme = {}) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);

export const withProviders = (component, initialEntries, theme = {}) => (
  <MemoryRouter initialEntries={initialEntries}>
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  </MemoryRouter>
);
