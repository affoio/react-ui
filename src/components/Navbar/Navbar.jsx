import styled from 'styled-components';
import colors from '../../colors';

const Navbar = styled.div`
  background-color: ${colors.white};
  width: 100%;
`;

export default Navbar;
