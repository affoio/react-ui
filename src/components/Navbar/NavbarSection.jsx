import styled, { css } from 'styled-components';
import colors from '../../colors';
import { getMultipliedIndent } from '../../theme/selectors';

const NavbarSection = styled.div`
  position: relative;
  ${({ theme }) => css`
    padding-bottom: ${getMultipliedIndent(theme, 6)};
    margin-bottom: ${getMultipliedIndent(theme, 6)};
  `}
  &:after {
    position: absolute;
    content: '';
    display: block;
    width: 32px;
    height: 3px;
    border-radius: 1.5px;
    background-color: ${colors.marigold[800]};
    bottom: 0;
    left: 60px;
  }
`;

export default NavbarSection;
