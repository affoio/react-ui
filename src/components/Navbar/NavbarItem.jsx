import React from 'react';

import styled, { css } from 'styled-components';
import { NavLink } from 'react-router-dom';
import colors from '../../colors';
import { getMultipliedIndent, getTypographyFontSizeStyles } from '../../theme/selectors';
import { NavbarContent } from './NavbarContent';
import { addStyleIfPropTruly } from '../../helpers/styledComponents';

const activeClassName = 'nav-item-active';

const activeStyles = `
    color: ${colors.indigoBlue[800]};
    background-color: ${colors.indigoBlue[100]};
  `;

export const NavbarItemStyled = styled.a.attrs({
  activeClassName,
})`
  height: 48px;
  display: flex;
  align-items: center;
  color: ${colors.black[500]};
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  
  ${({ theme }) => css`
    ${getTypographyFontSizeStyles(theme, 200)};
    padding-left: ${getMultipliedIndent(theme, 15)};
    padding-right: ${getMultipliedIndent(theme, 5)};

    ${NavbarContent} & {
      padding-left: ${getMultipliedIndent(theme, 19)};
    }
  `}

  &:hover,
  &.${activeClassName} {
    ${activeStyles}
  }

  ${addStyleIfPropTruly('active', activeStyles)};
`;

const NavbarItem = props => {
  const { external, to, as, children, onClick, active } = props;
  return (
    <NavbarItemStyled
      as={as || external ? 'a' : NavLink}
      to={to}
      onClick={onClick}
      href={external ? to : undefined}
      active={active}
    >
      {children}
    </NavbarItemStyled>
  );
};

export default NavbarItem;
