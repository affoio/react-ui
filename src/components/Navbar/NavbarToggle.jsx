import styled from 'styled-components';
import colors from '../../colors';
import { NavbarItemStyled } from './NavbarItem';

const NavbarToggle = styled(NavbarItemStyled).attrs({ as: 'div' })`
  ${({ isActive }) => isActive && `color: ${colors.indigoBlue[800]}`}
`;

export default NavbarToggle;
