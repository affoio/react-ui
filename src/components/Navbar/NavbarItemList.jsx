import React, { useState, useMemo } from 'react';
import { withRouter } from 'react-router-dom';
import { NavbarContent } from './NavbarContent';
import NavbarToggle from './NavbarToggle';
import NavbarItem from './NavbarItem';
import SelectArrow from '../Select/SelectArrow';

const NavbarItemList = ({ header, paths, location, onClick, withoutLinks }) => {
  const isActive = useMemo(() => !withoutLinks && !!paths.find(({ path }) => location.pathname === path), [
    location.pathname,
    paths,
    withoutLinks,
  ]);
  const [isOpen, changeOpen] = useState(isActive);
  const onClickHandler = onItemClick => () => {
    changeOpen(false);

    if (onItemClick) {
      onItemClick();
    } else if (onClick) {
      onClick();
    }
  };

  return (
    <>
      <NavbarToggle isActive={isActive} onClick={() => changeOpen(!isOpen)}>
        {header}
        <SelectArrow isOpen={isOpen} />
      </NavbarToggle>
      <NavbarContent isOpen={isOpen}>
        {paths.map(({ path, external, name, active, ...other }) => (
          <NavbarItem
            as={withoutLinks && 'div'}
            key={`${path}${name}`}
            to={path}
            external={external}
            onClick={onClickHandler(other.onClick)}
            active={active}
          >
            {name}
          </NavbarItem>
        ))}
      </NavbarContent>
    </>
  );
};

export default withRouter(NavbarItemList);
