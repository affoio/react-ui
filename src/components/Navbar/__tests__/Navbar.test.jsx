import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';
import NavbarItem from '../NavbarItem';
import NavbarSection from '../NavbarSection';
import { NavbarContent } from '../NavbarContent';
import NavbarItemList from '../NavbarItemList';
import { withProviders } from '../../../helpers/testHelpers';
import colors from '../../../colors';
import NavbarToggle from '../NavbarToggle';

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        withProviders(
          <Navbar>
            <NavbarSection>
              <NavbarItemList
                header="Dashboard"
                paths={[{ path: '/login', external: true, name: 'login' }, { path: '/forgot', name: 'forgot' }]}
              />
            </NavbarSection>
            <NavbarItem to="/singup">signup</NavbarItem>
          </Navbar>
        )
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('<NavbarItemList>', () => {
    it('toggles content', () => {
      const wrapper = mount(
        withProviders(
          <NavbarItemList
            header="Dashboard"
            paths={[{ path: '/login', external: true, name: 'login' }, { path: '/forgot', name: 'forgot' }]}
          />
        )
      );
      expect(toJSON(wrapper.find(NavbarContent).render())).toHaveStyleRule('display', 'none');

      wrapper.simulate('click');
      expect(toJSON(wrapper.find(NavbarContent).render())).toHaveStyleRule('display', 'block');
    });

    it('has active color when path matches', () => {
      const wrapper = mount(
        withProviders(
          <NavbarItemList
            header="Dashboard"
            paths={[{ path: '/login', external: true, name: 'login' }, { path: '/forgot', name: 'forgot' }]}
          />,
          ['/login']
        )
      );

      expect(toJSON(wrapper.find(NavbarToggle).render())).toHaveStyleRule('color', colors.indigoBlue[800]);
    });
  });
});
