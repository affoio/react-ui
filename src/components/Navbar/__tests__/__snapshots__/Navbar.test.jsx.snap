// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
.c0 {
  background-color: #ffffff;
  width: 100%;
}

.c7 {
  display: none;
}

.c8 {
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #8c8c8c;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  font-size: 14px;
  padding-left: 60px;
  padding-right: 20px;
}

.c6 .c8 {
  padding-left: 76px;
}

.c8:hover,
.c8.nav-item-active {
  color: #4718b4;
  background-color: #f6f3fb;
}

.c1 {
  position: relative;
  padding-bottom: 24px;
  margin-bottom: 24px;
}

.c1:after {
  position: absolute;
  content: '';
  display: block;
  width: 32px;
  height: 3px;
  border-radius: 1.5px;
  background-color: #ffca00;
  bottom: 0;
  left: 60px;
}

.c2 {
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #8c8c8c;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  font-size: 14px;
  padding-left: 60px;
  padding-right: 20px;
}

.c6 .c2 {
  padding-left: 76px;
}

.c2:hover,
.c2.nav-item-active {
  color: #4718b4;
  background-color: #f6f3fb;
}

.c5 {
  fill: #bfbfbf;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 4px;
  width: 24px;
  height: 24px;
}

.c4 {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
  -webkit-transition: -webkit-transform 0.2s linear;
  -webkit-transition: transform 0.2s linear;
  transition: transform 0.2s linear;
  will-change: transform;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      onClick={[Function]}
    >
      Dashboard
      <div
        className="c3"
      >
        <svg
          className="c4 c5"
          color="#bfbfbf"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      className="c6 c7"
    >
      <a
        className="c8"
        href="/login"
        onClick={[Function]}
        to="/login"
      >
        login
      </a>
      <a
        aria-current={null}
        className="c8"
        href="/forgot"
        onClick={[Function]}
      >
        forgot
      </a>
    </div>
  </div>
  <a
    aria-current={null}
    className="c8"
    href="/singup"
    onClick={[Function]}
  >
    signup
  </a>
</div>
`;
